src/Renderer/Camera.js

```diff
@@ -65,60 +80,119 @@ Camera.prototype.setRotation = function setRotation(rotation) {
-    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, sphere, tempSphere3d);
+    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, sphere, tmp.sphere3d);
 
-    localViewMatrix.multiplyMatrices(this._viewMatrix, tempMatrix);
-    frustum.setFromMatrix(localViewMatrix);
-    return frustum.intersectsSphere(tempSphere3d);
+    tmp.viewMatrix.multiplyMatrices(this._viewMatrix, tmp.matrix);
+    tmp.frustum.setFromMatrix(tmp.viewMatrix);
+    return tmp.frustum.intersectsSphere(tmp.sphere3d);
 };
 
 Camera.prototype.box3DSizeOnScreen = function box3DSizeOnScreen(box3d, matrixWorld) {
-    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, box3d, tempBox3d);
-    tempMatrix.premultiply(this._viewMatrix);
+    _prepareMatrix(matrixWorld, this._visibilityTestingOffset, box3d, tmp.box3d);
+    tmp.matrix.premultiply(this._viewMatrix);
+
+    return tmp.box3d.applyMatrix4(tmp.matrix);
+};
+
+Camera.prototype.box3DContourOnScreen = (function _box3DContourOnScreen() {
+    const points = [
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+        new THREE.Vector3(),
+    ];
+    return function box3DContourOnScreen(box3d, matrixWorld, clampToScreen = true) {
+        if (box3d.isEmpty()) {
+            return 0;
+        }
+
+        _prepareMatrix(matrixWorld, this._visibilityTestingOffset, box3d, tmp.box3d);
+        tmp.matrix.premultiply(this._viewMatrix);
+
+        // copy pasted / adapted from Box3.applyMatrix4 - begin
+        // NOTE: I am using a binary pattern to specify all 2^3 combinations below
+        points[0].set(tmp.box3d.min.x, tmp.box3d.min.y, tmp.box3d.min.z).applyMatrix4(tmp.matrix); // 000
+        points[1].set(tmp.box3d.min.x, tmp.box3d.min.y, tmp.box3d.max.z).applyMatrix4(tmp.matrix); // 001
+        points[2].set(tmp.box3d.min.x, tmp.box3d.max.y, tmp.box3d.min.z).applyMatrix4(tmp.matrix); // 010
+        points[3].set(tmp.box3d.min.x, tmp.box3d.max.y, tmp.box3d.max.z).applyMatrix4(tmp.matrix); // 011
+        points[4].set(tmp.box3d.max.x, tmp.box3d.min.y, tmp.box3d.min.z).applyMatrix4(tmp.matrix); // 100
+        points[5].set(tmp.box3d.max.x, tmp.box3d.min.y, tmp.box3d.max.z).applyMatrix4(tmp.matrix); // 101
+        points[6].set(tmp.box3d.max.x, tmp.box3d.max.y, tmp.box3d.min.z).applyMatrix4(tmp.matrix); // 110
+        points[7].set(tmp.box3d.max.x, tmp.box3d.max.y, tmp.box3d.max.z).applyMatrix4(tmp.matrix); // 111
+        // copy pasted / adapted from Box3.applyMatrix4 - end
+
+        for (const pt of points) {
+            // translate/scale to [0, width]x[0, height]
+            pt.x = this.width * (pt.x + 1) * 0.5;
+            pt.y = this.height * (1 - pt.y) * 0.5;
+            if (clampToScreen) {
+                pt.x = Math.min(this.width, Math.max(0, pt.x));
+                pt.y = Math.min(this.height, Math.max(0, pt.y));
```

Comment:
Here is a slightly simpler clipping example with a triangle instead of a projected box. (image from yaldex.com)
![triangle clipping](http://www.yaldex.com/games-programming/FILES/08fig33.gif ) 
Your code instead project orthogonally outside points to the screen boundary, which will result in the underestimated triangle (P5*, proj(P2,[P0*,P1*]), proj(P1,[P2*,P3*]), where proj(p,s) denotes the orthogonal projection of point p on segment s. Is it really what you want ?